openhgnn/models/SeHGNN.py

```diff
@@ -0,0 +1,294 @@
+            self.concat_project_layer = nn.Linear((num_feats + num_label_feats) * hidden, hidden)
+
+        if self.residual:
+            self.res_fc = nn.Linear(nfeat, hidden, bias=False)
+
+        def add_nonlinear_layers(nfeats, dropout, bns=False):
+            if bns:
+                return [
+                    nn.BatchNorm1d(hidden),
+                    nn.PReLU(),
+                    nn.Dropout(dropout)
+                ]
+            else:
+                return [
+                    nn.PReLU(),
+                    nn.Dropout(dropout)
+                ]
+
+        lr_output_layers = [
+            [nn.Linear(hidden, hidden, bias=not bns)] + add_nonlinear_layers(hidden, dropout, bns)
+            for _ in range(n_layers_2-1)]
+        self.lr_output = nn.Sequential(*(
+            [ele for li in lr_output_layers for ele in li] + [
+            nn.Linear(hidden, nclass, bias=False),
+            nn.BatchNorm1d(nclass)]))
+
+        if self.label_residual:
+            label_fc_layers = [
+                [nn.Linear(hidden, hidden, bias=not bns)] + add_nonlinear_layers(hidden, dropout, bns)
+                for _ in range(n_layers_3-2)]
+            self.label_fc = nn.Sequential(*(
+                [nn.Linear(nclass, hidden, bias=not bns)] + add_nonlinear_layers(hidden, dropout, bns) \
+                + [ele for li in label_fc_layers for ele in li] + [nn.Linear(hidden, nclass, bias=True)]))
+            self.label_drop = nn.Dropout(label_drop)
+
+        self.prelu = nn.PReLU()
+        self.dropout = nn.Dropout(dropout)
+        self.input_drop = nn.Dropout(input_drop)
+
+        self.reset_parameters()
+
+    def reset_parameters(self):
+        gain = nn.init.calculate_gain("relu")
+
+        for layer in self.feat_project_layers:
+            if isinstance(layer, Conv1d1x1):
+                layer.reset_parameters()
+        if self.label_feat_project_layers is not None:
+            for layer in self.label_feat_project_layers:
+                if isinstance(layer, Conv1d1x1):
+                    layer.reset_parameters()
+
+        if self.dataset != 'products':
+            nn.init.xavier_uniform_(self.concat_project_layer.weight, gain=gain)
+            nn.init.zeros_(self.concat_project_layer.bias)
+
+        if self.residual:
+            nn.init.xavier_uniform_(self.res_fc.weight, gain=gain)
+
+        for layer in self.lr_output:
+            if isinstance(layer, nn.Linear):
+                nn.init.xavier_uniform_(layer.weight, gain=gain)
+                if layer.bias is not None:
+                    nn.init.zeros_(layer.bias)
+
+        if self.label_residual:
+            for layer in self.label_fc:
+                if isinstance(layer, nn.Linear):
+                    nn.init.xavier_uniform_(layer.weight, gain=gain)
+                    if layer.bias is not None:
+                        nn.init.zeros_(layer.bias)
+
+    def forward(self, feats_dict, layer_feats_dict, label_emb):
```

Comment:
The input parameters of the forward function in OpenHGNN should be ``forward(self, hg, feat)``. Correct your model. Besides, why doesn't your model use APIs in DGL?